misc/io-internal/src/main/ballerina/print.bal

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+
+public type Printable any|error|PrintableRawTemplate;
+
+public type PrintableRawTemplate object {
+    public string[] & readonly strings;
+    public Printable[] insertions;
+};
+
+# Prints `any`, `error` or string templates(such as `The respective int value is ${val}`) value(s)
+# to the STDOUT followed by a new line.
+#
+# + values - The value(s) to be printed
+public isolated function println(Printable... values) {
+    string value = "";
+    foreach Printable printableValue in values {
+        PrintableClassImpl printable = new PrintableClassImpl(printableValue);
+        value = value.concat(printable.toString());
+    }
+
+    externPrintln(out(), java:fromString(value));
+}
+
+class PrintableClassImpl {
```

Comment:
Are these classes really needed? Can't the toString functions just be module-level functions that accept Printable and PrintableRawTemplate as params and return a string? Let's create an issue for ballerina/io if so.